ui/src/views/Scopes/ScopesetComparison/index.jsx

```diff
@@ -57,6 +71,13 @@ export default class ScopesetComparison extends Component {
       const scopesB = splitLines(scopeTextB);
       const scopesetDiff = this.getScopesetDiff(scopesA, scopesB);
       const cellColors = this.getCellColors(scopesetDiff);
+      const scopeParamA = scopesA.join('%');
+      const scopeParamB = scopesB.join('%');
+
+      this.props.history.push({
+        pathname: '/auth/scopes/compare',
+        search: `?qA=${scopeParamA}&qB=${scopeParamB}`,
+      });
```

Comment:
This logic seems to work nicely, however, it's a bit inconsistent with the way we parse/stringify query parameters in the rest of the site. Let's use the [`qs`](https://yarnpkg.com/en/package/qs) package to simplify things.

```jsx
import { parse, stringify } from 'qs';
```

We use the `qs` package in the following places:

```
src/components/ClientForm/index.jsx
src/utils/isThirdPartyLogin.js
src/views/Clients/ViewClients/index.jsx
src/views/Hooks/ListHooks/index.jsx
src/views/Provisioners/ViewWorkers/index.jsx
src/views/PulseMessages/index.jsx
src/views/ThirdPartyLogin/index.jsx
src/views/WorkerManager/WMViewWorkerPools/index.jsx
```

Feel free to use to look at these files for inspiration.

Reply:
I have implemented the `qs` package/lib. I found this quite neat because it allowed me to clean up the code a bit. In addition, I was able to incorporate some of the useful new syntaxes I have been learning about such as Object Destructuring and passing self-named object properties by themselves.

I also added the functionality that was suggested. The `componentDidMount()` will now additionally set the state for the `scopesetDiff` and `cellColors`.

The same `qs` implementation can be seen in both `ScopesetComparision` & `ScopesetExpander`. I had to change some variable naming due to namespace collisions.